### DISCUSSION TITLE
Diagnosing Pipeline Dry-run

### DISCUSSION THREAD

**Discussion by arnefromsa** [aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework#20] [Q&A] [answered]
I've created a new pipeline, and wanted to test it against a data piece, as documented in the End-to-End documentation.

I attempted to do a dryRun with the following error in the response : 
`{"message":"body/dryRunOptions/data/0 must be object"}`

The pipeline JSON is : 
`{
        "attributes": {
            "scope": "1",
            "type": "stationary combustion"
        },
        "name": "scope 1 stationary pipeline",
        "description": "data processing pipeline to calculate scope 1 stationary combustion",
        "transformer": {
            "transforms": [
                {
                    "index": 0,
                    "formula": "AS_TIMESTAMP(:date,'M/d/yyyy')",
                    "outputs": [
                        {
                            "description": "Timestamp of bill.",
                            "index": 0,
                            "key": "time",
                            "label": "Time",
                            "type": "timestamp"
                        }
                    ]
                },
                {
                    "index": 1,
                    "formula": "#heat_content_equation(:type,:therms,'co2',group='/')+#heat_content_equation(:type,:therms,'ch4',group='/')+#heat_content_equation(:type,:therms,'n2o',group='/')",
                    "outputs": [
                        {
                            "description": "therms",
                            "index": 0,
                            "key": "therms",
                            "label": "therms",
                            "type": "number",
                            "metrics":["scope1:stationary"]
                        }
                    ]
                }
            ],
            "parameters": [
                {
                    "index": 0,
                    "key": "date",
                    "type": "string"
                },
                {
                    "index": 1,
                    "key": "type",
                    "type": "string"
                },
                {
                    "index": 2,
                    "key": "therms",
                    "label": "therms",
                    "description": "therms",
                    "type": "number"
                }
            ]
        },
        "dryRunOptions": {
            "data": [
                "1/1/2021,natural gas,61500"
            ]
        }
    }`

Am I missing something in the JSON ?

**Comment by williamsia** [comment, marked as answer]
Hi @arnefromsa , the data should be array of object 

so in your case it should be 

```json
{
  "dryRunOptions": {
    "data": [
      {
        "date": "1/1/2021",
        "type": "natural gas",
        "therms": 61500
      }
    ]
  }
}

```

**Reply by arnefromsa** [reply]
Thanks @williamsia - worked perfectly now.
